src/components/CustomTable/StyledTableBody/StyledTableBody.js

```diff
@@ -27,12 +26,18 @@ const StyledTableBody = ({ columns, data, onRowClick }) => {
           classes={{ root: classes.root, hover: classes.hover }}
           onClick={() => onRowClick(row)}
         >
-          {columns.map((column) => (
-            <StyledTableCell
-              key={row.id + column.header}
-              value={get(row, column.field, "-")}
-            />
-          ))}
+          {columns.map((column) => {
+            const { header, field, cellRenderer } = column
+
+            return (
+              <React.Fragment key={row.id + header}>
+                {cellRenderer({
+                  cellData: get(row, field),
+                  rowData: row,
+                })}
```

Comment:
This should be a react component, change how you call this to 
`<CellRenderer cellData={get(row, field)} rowData={row}  />`